simtools/model/camera.py

```diff
@@ -0,0 +1,670 @@
+
+        Returns
+        -------
+        neighbours: array_like
+            Array of neighbour indices in a list for each pixel
+
+        '''
+
+        # First find the neighbours with the usual method and the original radius
+        # which does not allow for diagonal neighbours.
+        neighbours = self._findNeighbours(xPos, yPos, radius)
+        for i_pix, nn in enumerate(neighbours):
+            # Find pixels defined as edge pixels now
+            if len(nn) < 4:
+                # Go over all other pixels and search for ones which are adjacent
+                # but further than sqrt(2) away
+                for j_pix in range(len(xPos)):
+                    # No need to look at the pixel itself
+                    # nor at any pixels already in the neighbours list
+                    if j_pix != i_pix and j_pix not in nn:
+                        dist = np.sqrt((xPos[i_pix] - xPos[j_pix])**2 +
+                                       (yPos[i_pix] - yPos[j_pix])**2)
+                        # Check if this pixel is in the same row or column
+                        # and allow it to be ~1.68*diameter away (1.4*1.2 = 1.68)
+                        # Need to increase the distance because of the curvature
+                        # of the CHEC camera
+                        if ((abs(xPos[i_pix] - xPos[j_pix]) < rowColoumnDist or
+                            abs(yPos[i_pix] - yPos[j_pix]) < rowColoumnDist)
+                                and dist < 1.2*radius):
+                            nn.append(j_pix)
+
+        return neighbours
+
+    def getNeighbourPixels(self, pixels):
+        '''
+        Find adjacent neighbour pixels in cameras with hexagonal or square pixels.
+        Only directly adjacent neighbours are searched for, no diagonals.
+
+        Parameters
+        ----------
+        pixels: dictionary
+            The dictionary produced by the readPixelList method of this class
+
+        Returns
+        -------
+        neighbours: array_like
+            Array of neighbour indices in a list for each pixel
+
+        '''
+
+        logger.debug('Searching for neighbour pixels')
+
+        if pixels['funnelShape'] == 1 or pixels['funnelShape'] == 3:
+            neighbours = self._findNeighbours(
+                pixels['x'],
+                pixels['y'],
+                1.1*pixels['diameter']
+            )
+        elif pixels['funnelShape'] == 2:
+            # Distance increased by 40% to take into account gaps in the SiPM cameras
+            # Pixels in the same row/column can be 20% shifted from one another
+            # Inside find_adjacent_neighbour_pixels the distance is increased
+            # further for pixels in the same row/column to 1.68*diameter.
+            neighbours = self._findAdjacentNeighbourPixels(
+                pixels['x'],
+                pixels['y'],
+                1.4*pixels['diameter'],
+                0.2*pixels['diameter']
+            )
+
```

Comment:
Here, for example, it will break if funnelShape not in [1, 2, 3]. So either we verify it after reading the camera files, or we add else here to deal with other options.

Reply:
Yeah, verified when reading the file.